processor/src/chiplets/hasher/trace.rs

```diff
@@ -103,6 +103,32 @@ impl HasherTrace {
         self.node_index.push(index);
     }
 
+    /// Copies section of trace from the given range of start and end rows at the end of the trace.
+    pub fn copy_trace(&mut self, state: &mut [Felt; STATE_WIDTH], range: Range<usize>) {
+        for row in range {
+            let mut selectors: [Felt; NUM_SELECTORS] = Default::default();
+            for (col, selector) in selectors.iter_mut().enumerate() {
+                *selector = self.selectors[col][row];
+            }
+
+            let mut hasher_state: [Felt; STATE_WIDTH] = Default::default();
+            for (col, state) in hasher_state.iter_mut().enumerate() {
+                *state = self.hasher_state[col][row];
+            }
+
+            let node_index = self.node_index[row];
+            self.append_row(selectors, &hasher_state, node_index);
+        }
+
+        // TODO: remove this after the state is removed from the HasherLookup struct
+        // copy the latest hasher state to the provided state slice
+        for (state_col, hasher_col) in state.iter_mut().zip(self.hasher_state.iter()) {
```

Comment:
If we move `hsaher_state` variable outside of the loop, we might be able to just iterate over that (instead of `self.hasher_state`).